pipenv/utils.py

```diff
@@ -393,17 +394,23 @@ def clean_pkg_version(version):
 
 class HackedPythonVersion(object):
     """A Beautiful hack, which allows us to tell pip which version of Python we're using."""
+
+    PatchedSysVersion = namedtuple('PatchedSysVersion', ['major', 'minor', 'micro'])
+
     def __init__(self, python_version, python_path):
         self.python_version = python_version
         self.python_path = python_path
 
     def __enter__(self):
         os.environ['PIP_PYTHON_VERSION'] = str(self.python_version)
         os.environ['PIP_PYTHON_PATH'] = str(self.python_path)
+        self.backup_version_info = sys.version_info
```

Comment:
Is this being used by pip-tools or is it being added as a nice to have for use later?

Reply:
It's to reset `sys.version_info` outside of the context manager (in `__exit__`). We don't want that patched thing globally.

Reply:
Or was your question "Why are we patching `sys.version_info`? That's for pip to generate the right list of pep425 tags for wheel selection, i.e. Pick the right wheel for the right python version in the managed venv.

Reply:
Ok, that makes sense. Thanks for clarifying!